cirq-core/cirq/circuits/circuit.py

```diff
@@ -201,6 +202,11 @@ def unfreeze(self, copy: bool = True) -> cirq.Circuit:
             copy: If True and 'self' is a Circuit, returns a copy that circuit.
         """
 
+    @property
+    @abc.abstractmethod
+    def tags(self) -> tuple[Hashable, ...]:
+        """Returns a tuple of the Circuit's tags."""
+
```

Comment:
Please add an abstracmethod `with_tags` so that all derived classes have that method.
(it is already in FrozenCircuit).